language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/TypeGuardUtil.java

```diff
@@ -0,0 +1,122 @@
+        String symbolName = expr.name().text();
+        List<TypeSymbol> members = new ArrayList<>(symbol.memberTypeDescriptors());
+        String detail = "Destructure the variable " + symbolName + " with typeguard";
+        String snippet = "";
+        snippet += IntStream.range(0, members.size() - 1).mapToObj(value -> {
+            TypeSymbol bType = members.get(value);
+            String placeHolder = "\t${" + (value + 1) + "}";
+            return "if " + symbolName + " is " + CommonUtil.getModifiedTypeName(ctx, bType) + " {"
+                    + CommonUtil.LINE_SEPARATOR + placeHolder + CommonUtil.LINE_SEPARATOR + "}";
+        }).collect(Collectors.joining(" else ")) + " else {" + CommonUtil.LINE_SEPARATOR + "\t${"
+                + members.size() + "}" + CommonUtil.LINE_SEPARATOR + "}";
+
+        //create the text edit item to replace the field accession text.
+        List<TextEdit> textEdits = new ArrayList<>();
+        TextEdit textEdit = new TextEdit();
+        textEdit.setNewText("");
+        Position start = new Position(expr.lineRange().startLine().line(), expr.lineRange().startLine().offset());
+        Position end = new Position(expr.lineRange().endLine().line(), expr.lineRange().endLine().offset() + 1);
+        textEdit.setRange(new Range(start, end));
+        textEdits.add(textEdit);
+        CompletionItem completionItem =
+                TypeGuardCompletionItemBuilder.build(snippet, TYPE_GUARD_LABEL, detail, textEdits);
+        return new StaticCompletionItem(ctx, completionItem, StaticCompletionItem.Kind.OTHER);
+    }
+
+    /**
+     * Returns Typeguard completion items for union type variable symbols for the given
+     * expression node in the given completion context.
+     *
+     * @param ctx  Completion context
+     * @param expr expression node
+     * @return List of typeguard completion items
+     */
+    public static List<LSCompletionItem> getTypeGuardDestructedItems(BallerinaCompletionContext ctx,
+                                                                     ExpressionNode expr) {
+        List<LSCompletionItem> completionItems = new ArrayList<>();
+        if (TypeGuardUtil.isInFunctionBlockBodyContext(expr)) {
+            Optional<TypeSymbol> typeSymbols = expr.apply(new ContextTypeResolver(ctx));
+            completionItems.addAll(typeSymbols.stream().filter(typeSymbol -> {
+                return typeSymbol.typeKind() == TypeDescKind.UNION;
+            }).map(tSymbol -> {
+                return TypeGuardUtil.getTypeGuardDestructedItem((UnionTypeSymbol) tSymbol,
+                        ctx, (SimpleNameReferenceNode) expr);
+            }).collect(Collectors.toList()));
+        }
+
+        return completionItems;
+    }
+
+    /**
+     * Check if the given node is within a FunctionBodyBlockNode context.
+     *
+     * @param node
+     * @return
+     */
+    public static boolean isInFunctionBlockBodyContext(ExpressionNode node) {
```

Comment:
Modify this to accept the `FieldAccessExpression` along with the type capture suggestion above.

Reply:
Done!

Reply:
Shall we address the https://github.com/ballerina-platform/ballerina-lang/pull/30136#discussion_r618970949 as well? Passing the  `FieldAccessExpression` here?

Reply:
Done @nadeeshaan